atc/exec/task_step.go

```diff
@@ -440,8 +437,13 @@ func (action *TaskStep) registerOutputs(logger lager.Logger, repository *worker.
 
 		for _, mount := range volumeMounts {
 			if filepath.Clean(mount.MountPath) == filepath.Clean(outputPath) {
-				source := newTaskArtifactSource(mount.Volume)
-				repository.RegisterSource(worker.ArtifactName(outputName), source)
+				source := NewTaskArtifactSource(mount.Volume)
+				repository.RegisterSource(artifact.Name(outputName), source)
+
+				_, err := mount.Volume.InitializeArtifact(outputName, action.buildID)
```

Comment:
So rather than having a step that exists solely to proxy an artifact to the user, we initialize all task outputs as artifacts and just allow them to be downloaded via the API?

Does this mean that each build's task outputs now live for 12 hours? It seems like that could pile up and result in a lot more disk usage if so, assuming a build runs frequently enough.

Reply:
TBH I never even considered putting this in its own step (which is hilarious because I had to delete the old output artifact build step to make this change).

It wouldn't really help us with the lifecycle, but it would at least limit the number of `artifacts` that get created to only one-off builds.

Reply:
Hmm should we bring the artifact output step back, then? I wouldn't want to ship with all task outputs lasting for 12 hours.

Reply:
Yeah working on that now. It adds a lot better symmetry :)